modules/renter/dotsia/format.go

```diff
@@ -0,0 +1,390 @@
+/*
+Package dotsia defines the .sia format.
+
+Specification
+
+A .sia file is a gzipped tar archive containing one or more Files. Each
+File is a JSON representation of the File type defined in this package. For
+each file object header in the tar archive, only the Size field is
+populated. A File only contains metadata, not the actual file as uploaded
+to the Sia network; thus, metadata pertaining to the uploaded file, such as
+its path and mode bits, are specified inside the File, not the tar header.
+
+The first entry in the tar archive is a special metadata file. It is a JSON
+object corresponding to the Metadata type exported by this package. This
+object contains a version string, which indicates the version of the .sia
+format used. At this time, the .sia format has no promise of backwards or
+forwards compatibility, except that the version field will never be removed
+from the metadata object.
```

Comment:
as of 1.0 it would be nice to provide both backwards and forwards compatibility guarantees.

Backwards compatibility will be a requirement.